### PR TITLE
boards: sam0: deduplicate sam0 board makefiles

### PR DESCRIPTION
**Pull request by kaspar030**
This PR removes ~20 duplicated lines across 4 boards.

Please help testing:

- [x] samr21-xpro (@kaspar030)
- [x] saml21-xpro
- [ ] sodaq-autonomo
- [ ] samd21-xpro
- [x] arduino-zero

### CODE REVIEW

File: makefiles/boards/sam0.inc.mk [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# set default port depending on operating system
+PORT_LINUX ?= /dev/ttyACM0
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
+
+# setup serial terminal
+include $(RIOTMAKE)/tools/serial.inc.mk
+
+# Add board selector (USB serial) to OpenOCD options if specified.
+# Use /dist/tools/usb-serial/list-ttys.sh to find out serial number.
+#   Usage: SERIAL="ATML..." BOARD="samr21-xpro" make flash
```
Comment (aabadie) [inline makefiles/boards/sam0.inc.mk]: maybe use a generic `<BOARD>` instead of samr21-xpro

Comment (kaspar030) [inline makefiles/boards/sam0.inc.mk]: fixed!

**Comment by haukepetersen** [comment]
I can't help, I don't seem to have any of the other boards.

**Comment by dylad** [comment]
@kaspar030 Tested, nothing is broken on SAML21-XPRO board. You can remove it from the list.

**Comment by kaspar030** [comment]
> Minor comment, otherwise it looks good.

addressed!

> I can try on arduino-zero

IMHO if you can confirm that working, we can safely assume that sodaq-autonomo and sand21-xpro will work, too...

**Comment by haukepetersen** [comment]
Looking at the diffs again, I'd say we can just merge this even if we did not test for the last remaining boards explicitly - all of them had pretty much identical Makefile.includes, so IMHO we are fine with just testing a sub-set.

@aabadie: please merge if you feel your change requests are addressed!

**Comment by travisgriggs** [comment]
I'll test this on samd-xpro later today and report back

**Comment by aabadie** [comment]
> I'll test this on samd-xpro later today and report back

I tested on arduino-zero earlier today and it worked fine. Then I decided to merge since the CI was green. Hope this is fine !

**Comment by travisgriggs** [comment]
confirmed that sample programs compile and run on samd21-xpro

**Pull request merged**